sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -830,7 +830,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
 }
 
 trait LeafLike[T <: TreeNode[T]] { self: TreeNode[T] =>
-  override final def children: Seq[T] = Nil
+  override def children: Seq[T] = Nil
```

Comment:
@cloud-fan I am removing this `final` temporarily. If the approach of this PR is OK, I will add this back and refactor.